aten/src/ATen/native/Distance.cpp

```diff
@@ -145,7 +145,31 @@ Tensor _cdist_forward(const Tensor& x1, const Tensor& x2, const double p, c10::o
   return result;
 }
 
-Tensor _cdist_backward(const Tensor& grad, const Tensor& x1, const Tensor& x2, const double p, const Tensor& cdist) {
+Tensor _cdist_backward(const Tensor& grad, const Tensor& _x1, const Tensor& _x2, const double p, const Tensor& cdist) {
+  // Broadcasting might generate non-contiguous Tensors, so handle it before doing checks
+  int64_t c1 = _x1.size(-1);
+  int64_t c2 = _x2.size(-1);
+  int64_t r1 = _x1.size(-2);
+  int64_t r2 = _x2.size(-2);
+  auto dim1 = _x1.dim();
+  auto dim2 = _x2.dim();
+  IntArrayRef batch_tensor1(_x1.sizes().data(), dim1 - 2);
+  IntArrayRef batch_tensor2(_x2.sizes().data(), dim2 - 2);
+  std::vector<int64_t> expand_batch_portion = infer_size(batch_tensor1, batch_tensor2);
```

Comment:
I could tell you to use infer_size_dimvector here to avoid std::vector, and use DimVectors for you expand sizes below, and use MaybeOwned to avoid refcount bumps with your maybe expands, but cdist backward is so awful that none of this would matter.

Reply:
Ok leaving it like that for now then.